internetarchive/openlibrary
# Fixed the issue#9946

The issue#9946 was occurring due to "?mode=ebooks" being default at some point in time earlier. Since that is not the case anymore, according to the openlibrary/openlibrary/templates/type/edition/view.html, the editions_limit was being assigned the value 10 always. This prevented the default value of editions_limit to be None (which is then used in  openlibrary/openlibrary/templates/type/work/editions_database.html).

The change includes the 'None' to be one of the acceptable states, restoring the originally intended logic.


Closes #9946 


fix

### Technical

The implementations simply adds None into the set of acceptable states of the query parameter 'mode'.

### Testing

1. Go to the homepage and search for any book.
2. Click upon any book in the results.
3. Scroll down and look at the editions table.
4. If the "Showing X featured editions. View all Y editions?" text doesn't show anymore for cases where X is equal to Y (Img. 1), the issue has been fixed. And otherwise, it should show the aforementioned text (Img. 2).
### Screenshot

![image](https://github.com/user-attachments/assets/3d0db290-2314-439f-9d62-3ac19a73fa39)
Img. 1(when X==Y)

![image](https://github.com/user-attachments/assets/0ca13ef9-642e-49cb-9010-403fa00b0e18)
Img. 2 (when X!=Y)


### Stakeholders

@SivanC @jimchamp @cdrini
